src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -65,6 +65,23 @@ class ProxyService(proxyConfig: ProxyConfig,
           throw AccessTokenExpiredException() }
   }
 
+  def proxyLocalize(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName, request: HttpRequest, token: HttpCookiePair): Future[HttpResponse] = {
+    //check auth to see it
```

Comment:
Could `authCheck` be factored into a method, since it seems to be duplicated in `proxyNotebook`?

Reply:
Thinking about something like this (didn't test): 
```
def checkAuth(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName, action: NotebookClusterAction): Future[Unit] = {
  for {
    hasViewPermission <- authProvider.hasNotebookClusterPermission(userInfo, GetClusterStatus, googleProject.value, clusterName.string)
    hasActionPermission <- authProvider.hasNotebookClusterPermission(userInfo, action, googleProject.value, clusterName.string)
  } yield {
    if (!hasViewPermission) {
      throw ClusterNotFoundException(googleProject, clusterName)
    } else if (!hasActionPermission) {
      throw AuthorizationError(userInfo.userEmail)
    } else {
      ()
    }
  }
}
```

Reply:
I thought of factoring out the exception types in the signature too but got put off because it seemed unnecessarily complicated. This is definitely cleaner, thanks!